src/main/capi/prepared-c.cpp

```diff
@@ -120,6 +159,22 @@ static duckdb_state duckdb_bind_value(duckdb_prepared_statement prepared_stateme
 	return DuckDBSuccess;
 }
 
+duckdb_state duckdb_bind_parameter_index(duckdb_prepared_statement prepared_statement, idx_t *param_idx_out,
+                                         const char *name) {
+	auto wrapper = (PreparedStatementWrapper *)prepared_statement;
+	if (!wrapper || !wrapper->statement || wrapper->statement->HasError()) {
+		return DuckDBError;
+	}
+	auto &statement = wrapper->statement;
+	if (!statement->named_param_map.count(name)) {
+		return DuckDBError;
+	}
+	*param_idx_out = statement->named_param_map.at(name);
+	// Register that this value was provided with a name
+	wrapper->name_to_index[name] = *param_idx_out;
```

Comment:
This seems rather dirty - we have not actually bound the parameter yet, right?

Reply:
Yea I agree, I think I tried too hard to not change the existing API.
The existing API expects an index, and since I'd like to be able to verify that the value for a named parameter was actually provided with a name, I used this to do that.

So this is one of many good reasons to do the unifying of positional and named parameters you suggested